spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/ModuleSerializable.scala

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.utils.serializer
+
+import com.intel.analytics.bigdl.nn.Container
+
+import scala.collection.JavaConverters._
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Table
+import com.intel.analytics.bigdl.utils.serializer.DataConverter.RegularizerConverter
+import serialization.Bigdl.DataType
+import serialization.Bigdl.{AttrValue, BigDLModule, BigDLTensor}
+
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+
+/**
+ * [[ModuleSerializable]] trait inherits [[Loadable]] and [[Savable]]
+ * traits for module serialization
+ * it provides default implementation from [[ModuleSerializer]] using reflection
+ */
+trait ModuleSerializable extends Loadable with Savable{
+
+  override def loadModule[T: ClassTag](model : BigDLModule)
+                                      (implicit ev: TensorNumeric[T]) : ModuleData[T] = {
+    ModuleSerializer.loadModule(model)
```

Comment:
then why not put the default implementation insides the trait? Even package the implementations in an object, why the object need to extend the ModuleSerializable?